testsuite/support.py

```diff
@@ -112,8 +113,14 @@ def selftest(options):
     counters = report.counters
     checks = options.physical_checks + options.logical_checks
     for name, check, argument_names in checks:
+        python_version = None
         for line in check.__doc__.splitlines():
             line = line.lstrip()
+            match = SELFTEST_PY_REGEX.match(line)
+            if match:
+                python_version = tuple(map(int, match.groups()))
+            if python_version and sys.version_info < python_version:
+                continue
```

Comment:
leftover / dead code ?

Reply:
No. The check requires `3.10`, at least the way it's currently designed. This is needed to only run the `doctest` for Python 3.10 and up.

Reply:
https://github.com/PyCQA/pycodestyle/blob/11c41a490ad42132c31d0ec9a331111a3535ed6b/pycodestyle.py#L520-L527

Reply:
I would either (1) remove the doctests and put them in actual tests files or (2) remove the specializating in the implementation